app/src/main/java/com/infomaniak/swisstransfer/ui/screen/main/transferdetails/TransferDownload.kt

```diff
@@ -196,10 +197,20 @@ private suspend fun buildDownloadRequest(
             name = "SwissTransfer/$fileName.zip"
         }
     }
-    return DownloadManagerUtils.requestFor(
-        url = url,
-        nameWithoutProblematicChars = name,
-        mimeType = FileType.guessMimeTypeFromFileName(name),
-        userAgent = userAgent,
-    )
+    return runCatching {
+        DownloadManagerUtils.requestFor(
+            url = url,
+            nameWithoutProblematicChars = name,
+            mimeType = FileType.guessMimeTypeFromFileName(name),
+            userAgent = userAgent,
+        )
+    }.onFailure {
+        // On some Xiaomi devices, `setDestinationInExternalPublicDir` might need to create
+        // the Download directory, and might fail at it with an `IllegalStateException`.
+        SentryLog.w(TAG, "Failed to create the DownloadManager request", it)
+        // It's very likely that trying again will work, so we don't bother showing the error,
+        // and just let the user try again.
+    }.getOrNull()
```

Comment:
If it ever fails repeatedly, won't it just infinite loop now that it's not thrown because of the `repeatWhileActive` ?
To which extend are you sure about your "very likely" ?

Reply:
Shouldn't because the loop waits for user request (i.e. a click on the connected button) on ever iteration.

Reply:
See the line just before the call site here: https://github.com/Infomaniak/android-SwissTransfer/pull/455/files#diff-10a0103e82844f01d6274062d5c395aa58240b9cd1bebf6477d449ea842afccaR148-R157